tools/flang/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2019, Arm Ltd.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+add_executable(flang
+  main.cpp
+)
+
+set_target_properties(flang
+  PROPERTIES
+    COMPILE_FLAGS
+      -Wno-unused-parameter
+      -Wno-unused-variable
+      -Wno-maybe-uninitialized
+)
+
+llvm_map_components_to_libnames(LLVM_COMMON_LIBS
+  Support
+  Target
+  Option
+)
+
+target_link_directories(flang
```

Comment:
This requires cmake 3.13?

Reply:
Ah, my bad. This should be removed.

The right thing to do is to use `find_package(Clang REQUIRED CONFIG)`. I think this should live along the `find_package(LLVM ...)` which lives in `/CMakeLists.txt` One slightly tricky thing will be to get `find_package` to pick up the right clang, ideally by searching in the `${LLVM_DIR}/../clang`, or similar. I think this can in principle be controlled by a `HINTS` argument to `find_package`, but I couldn't make it do the right thing quickly.

Here's a link to [`find_package`](https://cmake.org/cmake/help/v3.9/command/find_package.html) for cmake 3.9, which is the minimum supported cmake version specified in the root CMakeLists.

Reply:
Done